qutebrowser/browser/downloads.py

```diff
@@ -600,7 +603,8 @@ def _set_filename(self, filename, *, force_overwrite=False):
                                                   os.path.expanduser('~'))
 
         self.basename = os.path.basename(self._filename)
-        last_used_directory = os.path.dirname(self._filename)
+        if remember_directory:
+            last_used_directory = os.path.dirname(self._filename)
```

Comment:
Strange, it fails locally when reverting the change... Are you by any chance running without `QUTE_BDD_WEBENGINE=true`? This issue only exists with the webengine backend, the test always passes with webkit.